src/theory/arith/nl/ext/monomial_check.cpp

```diff
@@ -311,8 +312,17 @@ int MonomialCheck::compareSign(
   {
     if (mvaoa.getConst<Rational>().sgn() != 0)
     {
-      Node lemma = av.eqNode(d_data->d_zero).impNode(oa.eqNode(d_data->d_zero));
-      d_data->d_im.addPendingArithLemma(lemma, InferenceId::NL_SIGN);
+      Node prem = av.eqNode(d_data->d_zero);
+      Node conc = oa.eqNode(d_data->d_zero);
+      Node lemma = prem.impNode(conc);
+      LazyCDProof* proof = nullptr;
+      if (d_data->d_proof)
```

Comment:
It's clearer if a condition like this is `isProofEnabled()`, can you define a method for checking this and use that here instead? This is the style we've been trying to follow throughout.